torch/autograd/profiler.py

```diff
@@ -481,6 +481,8 @@ def _start_kineto_trace(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         if not self.enabled:
             return
+        if torch.cuda.is_available():
+            torch.cuda.current_stream().synchronize()
```

Comment:
May I ask why needs to synchronize cuda when profiler exits?

Reply:
we might schedule some cuda kernels and exit the profiler before the kernels are executed on the device

Reply:
Gotit, thanks!